src/storage/childtries.md

```diff
@@ -1,4 +1,109 @@
+
+```rust
+fn kv_put(index: ObjectCount, who: &T::AccountId, value_type: &ValueType) {
+    let mut buf = Vec::new();
+		buf.extend_from_slice(b"exchildtr");
+		buf.extend_from_slice(&index.to_le_bytes()[..]);
+
+	let id = CHILD_STORAGE_KEY_PREFIX.into_iter()
+        .chain(b"default:")
+        .chain(T::Hashing::hash(&buf[..]).as_ref().into_iter())
+        .cloned()
+        .collect();
+    
+	who.using_encoded(|b| child::put(id.as_ref(), b, value_type));
+}
+```
+
+The code in [`kitchen/modules/child-trie`](https://github.com/substrate-developer-hub/recipes/tree/master/kitchen/modules/child-trie) demonstrates a minimal way of organizing the basic child-trie api methods (as done in [`polkadot/runtime/crowdfund`](https://github.com/paritytech/polkadot/blob/master/runtime/src/crowdfund.rs)). It separates out the generation of the child trie id from the index with a runtime method `id_from_index`.
+
+```rust
+pub fn id_from_index(index: ObjectCount) -> Vec<u8> {
+    let mut buf = Vec::new();
+    buf.extend_from_slice(b"exchildtr");
+    buf.extend_from_slice(&index.to_le_bytes()[..]);
+
+    CHILD_STORAGE_KEY_PREFIX
+        .into_iter()
+        .chain(b"default:")
+        .chain(Blake2Hasher::hash(&buf[..]).as_ref().into_iter())
+        .cloned()
+        .collect()
+}
+```
+
+This results in less code for each method:
+
+```rust
+pub fn kv_put(index: ObjectCount, who: &T::AccountId, value_type: ValueType) {
+    let id = Self::id_from_index(index);
+    who.using_encoded(|b| child::put(id.as_ref(), b, &value_type));
+}
+```
+
+## smpl-crowdfund <a name = "smplcrwd"></a>
+*[`kitchen/modules/smpl-crowdfund`](https://github.com/substrate-developer-hub/recipes/tree/master/kitchen/modules/smpl-crowdfund)*
+
+Child tries are useful for batch deletion of `(key, value)` pairs associated with a specific `trie_id`. This is relevant to the [polkadot/crowdfund](https://github.com/paritytech/polkadot/blob/master/runtime/src/crowdfund.rs) module, which tracks `(AccountId, BalanceOf<T>)` associated with a specific crowdfund. `BalanceOf<T>` represents the contributions of an `AccountId`. The identifier for each crowdfund is defined
+
+```rust
+type FundIndex = u32
+```
+
+With these three types, this storage item effectively manages `(FundIndex, AccountId, BalanceOf<T>)`. By maintaining a separate `child` for every `FundIndex`, this api allows for efficient batch deletions when crowdfunds are ended and dissolved.
+
+```rust
+// polkadot/runtime/crowdfund
+pub fn crowdfund_kill(index: FundIndex) {
+    let id = Self::id_from_index(index);
+    child::kill_storage(id.as_ref());
+}
+```
+
+It is still to be seen how child tries will be utilized in substrate runtimes, but it is a good candidate for storing data with strong correctness guarantees (all changes reflected in trie root), and a batching-oriented removal schedule based on some associated identity. Other operations like `get` and `put` can be made less expensive by also batching execution.
```

Comment:
```suggestion
It may be a good idea to consider using child tries if
* data associated with an identifier should be isolated to facilitate efficient batch removal (i.e. all the information associated with a given crowdfund is removed when the crowdfund is dissolved)
* a small proof needs to be stored on-chain or sent between chains (i.e. to verify inclusion of some data)
```